org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -188,7 +191,22 @@ public static String getPackageName(IJavaProject javaProject, URI uri) {
 			File file = ResourceUtils.toFile(uri);
 			//FIXME need to determine actual charset from file
 			String content = Files.toString(file, Charsets.UTF_8);
-			return getPackageName(javaProject, content);
+			if (content.isEmpty()) {
+				java.nio.file.Path path = Paths.get(uri);
+				java.nio.file.Path parent = path;
+				while (parent.getParent() != null) {
+					parent = parent.getParent();
+					String name = parent.getName(parent.getNameCount() - 1).toString();
+					if (SRC.equals(name)) {

```

Comment:
We already use it. See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java#L158